dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoJedisClient.java

```diff
@@ -3255,8 +3269,74 @@ public Builder withDiscoveryClient(DiscoveryClient client) {
             return this;
         }
 
+        public Builder withShadowClusterName(String shadowCluster) {
+            shadowClusterName = shadowCluster;
+            return this;
+        }
+
+        public Builder withShadowHostSupplier(HostSupplier shadowHostSupplier) {
```

Comment:
Do we need two Host Suppliers? One for the primary and one for the shadow? or is it usually the case that we will have just one host supplier?